airflow/providers/google/cloud/hooks/pubsub.py

```diff
@@ -545,8 +549,20 @@ def acknowledge(
         :param metadata: (Optional) Additional metadata that is provided to the method.
         :type metadata: Sequence[Tuple[str, str]]]
         """
+
         if not project_id:
             raise ValueError("Project ID should be set.")
+
+        if ack_ids is not None and messages is None:
```

Comment:
The idea is that passing both of those arguments in function call is prohibited. Defaulting to empty list/tuple and using non-emptiness check rather than `is not None` would miss cases when users pass both arguments, one of which happens to be an empty collection, when they are not supposed to do that in the first place.

Reply:
Yep. I understand it - so you are saying that passing an empty list is OK. How then it should behave if you pass empty ack_id or messages ? I think this should fail as well.

Reply:
I think we should allow passing an empty list, because the `pull` method can return an empty list as well - `acknowledge` should be able to gracefully accept any valid `pull` return value. We could check for empty list and return early, or just pass the list to the underlying API and expose its behaviour without additional checks (as it is now).